src/shared/listener-collection.ts

```diff
@@ -13,57 +12,51 @@
  * The event type is extracted from the target's `on${Type}` property (eg.
  * `HTMLElement.onkeydown` here) If there is no such property, the type defaults
  * to `Event`.
- *
- * @template {EventTarget} Target
- * @template {string} Type
- * @typedef {`on${Type}` extends keyof Target ?
- *   Target[`on${Type}`] extends ((...args: any[]) => void)|null ?
- *     Parameters<NonNullable<Target[`on${Type}`]>>[0]
- *  : Event : Event} EventType
  */
+type EventType<
+  Target extends EventTarget,
+  Type extends string
+> = `on${Type}` extends keyof Target
+  ? Target[`on${Type}`] extends ((...args: any[]) => void) | null
+    ? Parameters<NonNullable<Target[`on${Type}`]>>[0]
+    : Event
+  : Event;
 
 /**
  * Utility that provides a way to conveniently remove a set of DOM event
  * listeners when they are no longer needed.
  */
 export class ListenerCollection {
+  private _listeners: Map<symbol, Listener>;
+
   constructor() {
-    /** @type {Map<Symbol, Listener>} */
     this._listeners = new Map();
   }
 
   /**
    * Add a listener and return an ID that can be used to remove it later
-   *
-   * @template {string} Type
-   * @template {EventTarget} Target
-   * @param {Target} eventTarget
-   * @param {Type} eventType
-   * @param {(event: EventType<Target, Type>) => void} listener
-   * @param {AddEventListenerOptions} [options]
    */
-  add(eventTarget, eventType, listener, options) {
-    eventTarget.addEventListener(
-      eventType,
-      /** @type {EventListener} */ (listener),
-      options
-    );
+  add<Type extends string, Target extends EventTarget>(
+    eventTarget: Target,
+    eventType: Type,
+    listener: (event: EventType<Target, Type>) => void,
+    options?: AddEventListenerOptions
+  ) {
+    eventTarget.addEventListener(eventType, listener as EventListener, options);
     const symbol = Symbol();
     this._listeners.set(symbol, {
       eventTarget,
       eventType,
       // eslint-disable-next-line object-shorthand
-      listener: /** @type {EventListener} */ (listener),
+      listener: listener as EventListener,
     });
     return symbol;
   }
 
   /**
    * Remove a specific listener.
-   *
-   * @param {Symbol} listenerId
    */
-  remove(listenerId) {
+  remove(listenerId: symbol) {
```

Comment:
`symbol` is a primitive type like `string` or `number`, so the preferred type name is lower-case. This was found by the linter.

Reply:
Ah! I think I woulda tried `Symbol` myself, yep.